Support StorageArea.onChanged (#16873)
mdn/content: files/en-us/mozilla/add-ons/webextensions/api/storage/session/index.md

tags:
  - API
  - Add-ons
  - Extensions
  - Non-standard
  - Property
  - Reference
  - Storage
  - WebExtensions
  - session
browser-compat: webextensions.api.storage.session
---
{{AddonSidebar()}}

Represents the `session` storage area. Items in `session` storage are stored in memory and are not persisted to disk.

The browser may restrict the amount of data that an extension can store in the session storage area. For example, in Chrome, an extension is limited to storing 1MB of data in this storage area.

When the browser stops, all session storage is cleared. When the extension is uninstalled, its associated session storage is cleared.

## Methods

The `session` object implements the methods defined on the {{WebExtAPIRef("storage.StorageArea")}} type:

- {{WebExtAPIRef("storage.StorageArea.get()", "storage.session.get()")}}
  - : Retrieves one or more items from the storage area.
- {{WebExtAPIRef("storage.StorageArea.getBytesInUse()", "storage.session.getBytesInUse()")}}
  - : Gets the amount of storage space (in bytes) used for one or more items in the storage area.
- {{WebExtAPIRef("storage.StorageArea.set()", "storage.session.set()")}}

>   - : Stores one or more items in the storage area. If the item exists, its value is updated. When you set a value, if they have been implemented, the {{WebExtAPIRef("storage.onChanged")}} and {{WebExtAPIRef("storage.StorageArea.onChanged", "storage.session.onChanged")}} events fire.

Reviewer: Remove "if they have been implemented". This is covered by the BCD table.

It's not necessary to mention onChanged at every `set` method; the event is fired whenever the storage is modified, whether with set, remove or clear.

Author: @Rob--W I was thinking it makes sense to mention these events on the basis that we don't know what order someone's reading the documentation in.

>   - : Stores one or more items in the storage area. If the item exists, its value is updated. When you set a value, if they have been implemented, the {{WebExtAPIRef("storage.onChanged")}} and {{WebExtAPIRef("storage.StorageArea.onChanged", "storage.session.onChanged")}} events fire.

Author: ```suggestion
  - : Stores one or more items in the storage area. If the item exists, its value is updated. When you store a value, the {{WebExtAPIRef("storage.onChanged")}} and {{WebExtAPIRef("storage.StorageArea.onChanged", "storage.session.onChanged")}} events fire.
```
